### PR TITLE
When generating department reports only generate for current year

### PR DESCRIPTION
**Pull request by AricStewart**

### CODE REVIEW

File: functions/volunteer.inc [hunk 1]
```diff
@@ -602,8 +602,12 @@ SQL;
 }
 
 
-function volunteers_in_dept($department)
+function volunteers_in_dept($department, $event = null)
 {
+    if ($event === null) {
+        $event = current_eventID();
```
Comment (mshappe) [inline functions/volunteer.inc]: In PHP, can the initializer in the argument list be a method call? That is, can you just say

```php
function volunteers_in_dept($department, $event = current_eventID())
```

and skip a step?

File: modules/volunteers/admin/pages/pre.inc [hunk 2]
```diff
@@ -44,8 +44,9 @@ if (!empty($_REQUEST)) {
         if (isset($_SESSION['IS_ADMIN']) || $_SESSION['IS_VOLUNTEERS']) {
             $dept = $updateData['dept_report'];
             $name = $updateData['dept_name'];
-            $report = volunteers_in_dept(intval($dept));
-            output_csv($report, "department_report_".$name.".csv");
+            $event = current_eventID();
```
Comment (mshappe) [inline modules/volunteers/admin/pages/pre.inc]: Redundant with the way you're already initializing this in `volunteers_in _dept`.

Also, what if I really **want** a list of vols across all years?

File: functions/volunteer.inc (hunk 1 above)
Comment (AricStewart) [reply functions/volunteer.inc]: I do not believe so. Or at least a few minutes of web searching did not turn up a way to do that. And quick test programs do not work.

File: modules/volunteers/admin/pages/pre.inc (hunk 2 above)
Comment (AricStewart) [reply modules/volunteers/admin/pages/pre.inc]: Yes I am being redundant but part of me is waiting for the request to run reports or do actions on historical data. So as long as I keep things able to do that it makes it easier to fufill that potential future request.

Comment (AricStewart) [reply modules/volunteers/admin/pages/pre.inc]: as for the multiple years that makes sense.  I will work it out to allow for that.

Comment (AricStewart) [reply modules/volunteers/admin/pages/pre.inc]: Patch updated to allow for multiple years or all years. So when/if we add UI for that we can make it work.

File: functions/volunteer.inc (hunk 1 above)
Comment (mshappe) [reply functions/volunteer.inc]: 👍

File: modules/volunteers/admin/pages/pre.inc (hunk 2 above)
Comment (mshappe) [reply modules/volunteers/admin/pages/pre.inc]: 👍

**Comment by mshappe** [comment]
I'm still not authorized to merge, so that'll have to be @tskeeley

**Comment by AricStewart** [comment]
> I'm still not authorized to merge, so that'll have to be @tskeeley

You should be able to now. try again.

**Comment by seanbseanb** [comment]
Can I drop this one into prod, or will it conflict with other commits?

**Pull request merged**